### PR TITLE
Add clearsSelectionOnViewWillAppear

### PR DESCRIPTION
**Pull request by 3lvis**
Fixes https://github.com/UseSweet/SweetUIKit/issues/61

### CODE REVIEW

File: Sources/SweetTableController.swift [hunk 1]
```diff
@@ -24,6 +26,16 @@
             self.addConstraints()
         }
 
+        open override func viewWillAppear(_ animated: Bool) {
+            super.viewWillAppear(animated)
+
+            if self.clearsSelectionOnViewWillAppear {
```
Comment (elland) [inline Sources/SweetTableController.swift:32]: I don't follow. Isn't that what `clearsSelection...` does already? Also why is it defaulting to `false` now? 🤔

Comment (3lvis) [reply Sources/SweetTableController.swift:32]: `clearsSelectionOnViewWillAppear` is available for UITableViewController subclasses, we don't have that. And it also defaults to false, we can make it default to true, though.

Comment (elland) [reply Sources/SweetTableController.swift:32]: >  is available for UITableViewController

😅 right!

Does the end-result of your implementation looks the same as UIKits in the end? 😄

I'd prefer if it defaults to true. I've yet to see the case for preserving selection when popping back.

Comment (3lvis) [reply Sources/SweetTableController.swift:32]: Yeah, I agree, should be true.

Comment (3lvis) [reply Sources/SweetTableController.swift:32]: Regarding the result, I tried in a different app. It went fine.

Comment (elland) [reply Sources/SweetTableController.swift:32]: 👏🏼👏🏼👷🏼‍♂️💪🏼

**Comment by elland** [comment]
💥

**Pull request merged**